File: MobiTaskClient/app/src/main/java/com/task/mobi/fragment/LoginFragment.java
package com.task.mobi.fragment;

import android.content.Context;
import android.content.Intent;
import android.net.wifi.WifiManager;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.EditText;

import com.google.gson.Gson;
import com.google.gson.JsonObject;
import com.google.gson.reflect.TypeToken;
import com.task.mobi.BuildConfig;
import com.task.mobi.R;
import com.task.mobi.activity.MainActivity;
import com.task.mobi.data.UserData;
import com.task.mobi.utils.Const;
import com.task.mobi.utils.MacUtils;

import java.lang.reflect.Type;

import butterknife.BindView;
import butterknife.OnClick;
import okhttp3.MediaType;
import okhttp3.RequestBody;
import retrofit2.Call;

/**
 * Created by Neeraj Narwal on 5/5/17.
 */
public class LoginFragment extends BaseFragment {

    public View view;

    @BindView(R.id.emailET)
    EditText emailET;

    @BindView(R.id.passwordET)
    EditText passwordET;

    @BindView(R.id.rememberCB)
    CheckBox rememberCB;

    private Call<JsonObject> loginCall;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        if (view == null) {
            view = inflater.inflate(R.layout.fg_login, null);
        }
//        Utils.setStatusBarTranslucent(baseActivity, true);
        return view;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (BuildConfig.DEBUG) {
            emailET.setText("guru.p@pugmarks.com");
            passwordET.setText("admin123");
        }
        if (store.getBoolean(Const.REMEMBER_ME)) {
            rememberCB.setChecked(true);
            emailET.setText(store.getString(Const.REMEMBER_EMAIL));
            passwordET.setText(store.getString(Const.REMEMBER_PASS));
        }
    }

    @OnClick(R.id.loginBT)
    public void submit() {
        if (validate()) {
            login();
        }
    }

    @OnClick(R.id.forgotTV)
    public void resetPassword() {
        baseActivity.getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.frame_container_lf, new ForgotPasswordFragment())
                .addToBackStack(null)
                .commit();
    }

    private void login() {
        WifiManager wifiManager = (WifiManager) getActivity().getApplicationContext().getSystemService(Context.WIFI_SERVICE);
        String mac = MacUtils.macAddress(wifiManager);

        RequestBody email = RequestBody.create(MediaType.parse("text/plain"), getText(emailET));
        RequestBody password = RequestBody.create(MediaType.parse("text/plain"), getText(passwordET));
        RequestBody deviceId = RequestBody.create(MediaType.parse("text/plain"), mac + ":U");
        loginCall = apiInterface.login(email, password, deviceId);
        apiManager.makeApiCall(loginCall, this);
    }

    private boolean validate() {
        if (getText(emailET).isEmpty()) {
            showToast("Please enter email", true);
        } else if (getText(passwordET).isEmpty()) {
            showToast("Please enter password", true);
        } else
            return true;
        return false;
    }

    @Override
    public void onSuccess(Call call, Object object) {
        super.onSuccess(call, object);
        if (loginCall == call) {
            if (rememberCB.isChecked()) {
                store.setBoolean(Const.REMEMBER_ME, true);
                store.saveString(Const.REMEMBER_EMAIL, getText(emailET));
                store.saveString(Const.REMEMBER_PASS, getText(passwordET));
            }
            JsonObject jsonObject = (JsonObject) object;
            JsonObject data = jsonObject.getAsJsonObject("data");
            Type objectType = new TypeToken<UserData>() {
            }.getType();
            UserData userData = new Gson().fromJson(data, objectType);
            store.saveString(Const.SESSION_ID, userData.sessionId);
            store.saveString(Const.USER_ID, userData.id);
            store.saveUserData(Const.USER_DATA, userData);
            gotoMainAct();
        }
    }

    private void gotoMainAct() {
        startActivity(new Intent(baseActivity, MainActivity.class));
        baseActivity.finish();
    }
}

File: MobiTaskAdmin/app/src/main/java/com/task/mobiadmin/fragment/login_signup/LoginFragment.kt
package com.task.mobiadmin.fragment.login_signup

import android.content.Context
import android.content.Intent
import android.net.wifi.WifiManager
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.google.gson.Gson
import com.google.gson.JsonObject
import com.google.gson.reflect.TypeToken
import com.task.mobiadmin.BuildConfig
import com.task.mobiadmin.R
import com.task.mobiadmin.activity.MainActivity
import com.task.mobiadmin.data.UserData
import com.task.mobiadmin.fragment.BaseFragment
import com.task.mobiadmin.utils.Const
import com.task.mobiadmin.utils.MacUtils
import kotlinx.android.synthetic.main.fg_login.*
import okhttp3.MediaType
import okhttp3.RequestBody
import retrofit2.Call

/**
 * Created by Neeraj Narwal on 5/5/17.
 */
class LoginFragment : BaseFragment() {

    private var loginCall: Call<JsonObject>? = null

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater!!.inflate(R.layout.fg_login, null)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        if (BuildConfig.DEBUG) {
            emailET.setText("neeraj.n@pugmarks.com")
            passwordET.setText("admin123")
        }
        if (store.getBoolean(Const.REMEMBER_ME)) {
            rememberCB.isChecked = true
            emailET.setText(store.getString(Const.REMEMBER_EMAIL))
            passwordET.setText(store.getString(Const.REMEMBER_PASS))
        }
        backIV.setOnClickListener { baseActivity.onBackPressed() }
        loginBT.setOnClickListener { if (validate()) login() }
        forgotTV.setOnClickListener {
            baseActivity.supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.frame_container_ls, ForgotPasswordFragment())
                    .addToBackStack(null)
                    .commit()
        }
    }

    private fun login() {
        val wifiMan = activity.applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
        val mac = MacUtils.macAddress(wifiMan)
        val email = RequestBody.create(MediaType.parse("text/plain"), getText(emailET))
        val password = RequestBody.create(MediaType.parse("text/plain"), getText(passwordET))
        val deviceId = RequestBody.create(MediaType.parse("text/plain"), mac)
        loginCall = apiInterface.login(email, password, deviceId)
        apiManager.makeApiCall(loginCall, this)
    }

    private fun validate(): Boolean {
        when {
            getText(emailET).isEmpty() -> showToast("Please enter email", true)
            getText(passwordET).isEmpty() -> showToast("Please enter password", true)
            else -> return true
        }
        return false
    }

    override fun onSuccess(call: Call<*>, `object`: Any) {
        super.onSuccess(call, `object`)
        if (loginCall === call) {
            if (rememberCB.isChecked) {
                store.setBoolean(Const.REMEMBER_ME, true)
                store.saveString(Const.REMEMBER_EMAIL, getText(emailET))
                store.saveString(Const.REMEMBER_PASS, getText(passwordET))
            }
            val jsonObject = `object` as JsonObject
            val data = jsonObject.getAsJsonObject("data")
            val objectType = object : TypeToken<UserData>() {}.type
            val userData = Gson().fromJson<UserData>(data, objectType)
            store.saveString(Const.SESSION_ID, userData.sessionId)
            store.saveString(Const.USER_ID, userData.id)
            store.saveUserData(Const.USER_DATA, userData)
            gotoMainAct()
        }
    }

    private fun gotoMainAct() {
        startActivity(Intent(baseActivity, MainActivity::class.java))
        baseActivity.finish()
    }
}
